File: 7568/7568.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   7568.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/10/12 16:35:14 by jihoolee          #+#    #+#             */
/*   Updated: 2021/10/12 16:47:00 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>

using namespace std;

int	get_bulk_rank(int idx, int cases, int *weight, int *height)
{
	int	bigger = 0;

	for (int i = 0; i < cases; i++)
		if (weight[i] > weight[idx] && height[i] > height[idx])
			bigger++;
	return (bigger + 1);
}

int	main(void)
{
	int	cases;
	int	*weight;
	int	*height;

	cin >> cases;
	weight = new int[cases];
	height = new int[cases];
	for (int i = 0; i < cases; i++)
		cin >> weight[i] >> height[i];
	for (int i = 0; i < cases; i++)
		cout << get_bulk_rank(i, cases, weight, height) << ' ';
	return (0);
}

File: 7568/7568.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    7568.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/10/12 16:48:19 by jihoolee          #+#    #+#              #
#    Updated: 2021/10/12 17:04:05 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_bulk_rank(idx, w_h_lst):
	bigger = 0
	for weight, height in w_h_lst:
		if (weight > w_h_lst[idx][0] and height > w_h_lst[idx][1]):
			bigger += 1
	return (bigger + 1)

cases = int(input())
weight_height_lst = [list(map(int, input().split())) for _ in range(cases)]

bulk_rank = [get_bulk_rank(i, weight_height_lst) for i in range(cases)]
print(*bulk_rank)
